### PR TITLE
doc: Build openapi docs from release versions of dsp-api and dsp-ingest (DEV-3443)

### PR DESCRIPTION
**Pull request by seakayone**
- **Use double quote to prevent globbing**
- **Generate openapi from dsp-ingest and dsp-api from release versions when building the documentation**

Needs:
- https://github.com/dasch-swiss/dsp-ingest/pull/219
- https://github.com/dasch-swiss/dsp-api/pull/3254

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,4 +5,4 @@ mkdocs-include-markdown-plugin==6.0.4
 mkdocs-git-revision-date-localized-plugin==1.2.4
 mike==2.0.0
 neoteroi-mkdocs==1.0.5
-
+pymdown-extensions
```
Comment (jnussbaum) [inline requirements.txt:8]: What is this used for? 

I saw on https://facelessuser.github.io/pymdown-extensions/usage_notes/ that this does some of magic under the hood. How confident can we be that it doesn't break anything?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -27,6 +27,13 @@ jobs:
           sudo apt-get install graphviz
       - name: build graphviz figures
         run: make graphvizfigures
+      - name: Install just
```
Comment (jnussbaum) [inline .github/workflows/main.yml:30]: Could you add in the Readme how to install these 2 programs, if someone wants to serve the docs locally?

File: Makefile [hunk 3]
```diff
@@ -16,10 +16,17 @@ update-submodules: ## grab the current documentation from each connected repo
...
 	rm -rf ./docs/openapi
 	mkdir -p ./docs/openapi/
+	rm -rf ./docs/03-endpoints/generated-openapi
+	mkdir -p ./docs/03-endpoints/generated-openapi
+	## generate openapi yml freshly from the code
+	(cd ./dsp/dsp-api && just dog)
+	(cd ./dsp/dsp-ingest && just dog)
+	## mkdocs cannot resolve relative paths to the generate openapi yml, we need to copy them to the right location 
 	cp -r ./dsp/dsp-ingest/docs/openapi/*.yml ./docs/openapi/
+	cp -r ./dsp/dsp-api/docs/03-endpoints/generated-openapi/*.yml ./docs/03-endpoints/generated-openapi
```
Comment (jnussbaum) [inline Makefile:29]: Is this the right place for these files? If I serve the docs locally, and search for them in my browser, I cannot find them.

File: Makefile [hunk 4]
```diff
@@ -16,10 +16,17 @@ update-submodules: ## grab the current documentation from each connected repo
...
+openapi-update: 
 	rm -rf ./docs/openapi
 	mkdir -p ./docs/openapi/
+	rm -rf ./docs/03-endpoints/generated-openapi
+	mkdir -p ./docs/03-endpoints/generated-openapi
+	## generate openapi yml freshly from the code
+	(cd ./dsp/dsp-api && just dog)
+	(cd ./dsp/dsp-ingest && just dog)
+	## mkdocs cannot resolve relative paths to the generate openapi yml, we need to copy them to the right location 
 	cp -r ./dsp/dsp-ingest/docs/openapi/*.yml ./docs/openapi/
```
Comment (jnussbaum) [inline Makefile:28]: When serving the docs locally, when I navigate to http://localhost:8000/latest/Dsp-Ingest/api-endpoints-projects/, the title shows an outdated version: dsp-ingest-projects v0.9.1-dirty

Now that the openapi docs are every time generated fresh, I was surprised to still see the old version

File: Makefile [hunk 5]
```diff
@@ -16,10 +16,17 @@ update-submodules: ## grab the current documentation from each connected repo
...
+	cp -r ./dsp/dsp-api/docs/03-endpoints/generated-openapi/*.yml ./docs/03-endpoints/generated-openapi
 
 .PHONY: build
 build: ## build docs into the local 'site' folder
```
Comment (jnussbaum) [inline Makefile:32]: Another observation, not belonging to this PR: `docs/openapi/openapi-projects.yml` contains a little mistake:

```
- url: https://ingest.dev.dasch.swiss
  description: Production server
```

**Review by jnussbaum — COMMENTED** [review]
Thanks for your PR :-)
Unfortunately, I'm not sure if everything works as expected, see my comments.

File: Makefile (hunk 4 above)
Comment (seakayone) [reply Makefile:28]: The latest released version is v0.9.1 , the dirty is due to the fact the the changes I had to to in dsp-ingest/dsp-api are not yet merged and released.

It should be the right version once these changes are merged.

File: requirements.txt (hunk 1 above)
Comment (seakayone) [reply requirements.txt:8]: Tested it locally and it works. Without this extension I had errors.

File: Makefile (hunk 3 above)
Comment (seakayone) [reply Makefile:29]: Yes it is.

File: .github/workflows/main.yml (hunk 2 above)
Comment (seakayone) [reply .github/workflows/main.yml:30]: => https://github.com/dasch-swiss/dsp-docs/pull/114/commits/2ff0e8698d15c2c48560925533298da6fe1bee12

File: Makefile (hunk 3 above)
Comment (seakayone) [reply Makefile:29]: These are embedded into the `[OAD …]` markdown directives in dsp-ingest and dsp-api. For example: https://github.com/dasch-swiss/dsp-ingest/blob/main/docs/api-endpoints-projects.md?plain=1#L1

The relative path in above markdown document point to the location these files are copied to with the above lines.

Comment (seakayone) [reply Makefile:29]: The are not to be meant to be served directly.

File: requirements.txt (hunk 1 above)
Comment (seakayone) [reply requirements.txt]: I have given it another try and now it seems like this dependency is not needed really.

 [Removed pymdown-extensions](https://github.com/dasch-swiss/dsp-docs/pull/114/commits/a44a105592e815aba8dfb57d48fb8a780461854f).

File: Makefile (hunk 3 above)
Comment (jnussbaum) [reply Makefile:29]: For dsp-ingest, this works on my machine. But not so for dsp-api, see my screenshot:
Screenshot 2024-05-28 at 12 07 26

Comment (jnussbaum) [reply Makefile:29]: Or is this `OAD` link on my screenshot intended? If I click on it, it downloads a YAML file, but I guess this is not intended.

Comment (seakayone) [reply Makefile:29]: For this particular page to work you should update the dsp-api submodule to the linked branch.

This fixed it: https://github.com/dasch-swiss/dsp-api/pull/3254/files#diff-87d30c8cf1729db483920a0de9e85c268e51b04eb6b451e98f8a6202b430d39bL6

File: README.md [hunk 6]
```diff
@@ -53,6 +53,15 @@ Install it with
 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
 ```
 
+### Just
```
[the remark this answers is not in the document]
Comment (seakayone) [reply README.md:56]: and => https://github.com/dasch-swiss/dsp-docs/pull/114/commits/07fd0aee9becead52c80c9abaa8edcc2273975b4 (adds temurin)

**Review by BalduinLandolt — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (jnussbaum) [reply Makefile:29]: Thanks for the explanation. With the linked branch, it works 👍

**Review by jnussbaum — APPROVED** [review]

**Pull request merged**